documents/coredocument.go

```diff
@@ -651,14 +651,20 @@ func (cd *CoreDocument) Timestamp() (time.Time, error) {
 }
 
 // AddAttributes adds a custom attribute to the model with the given value. If an attribute with the given name already exists, it's updated.
-func (cd *CoreDocument) AddAttributes(attrs ...Attribute) (*CoreDocument, error) {
+func (cd *CoreDocument) AddAttributes(ca CollaboratorsAccess, prepareNewVersion bool, attrs ...Attribute) (*CoreDocument, error) {
 	if len(attrs) < 1 {
 		return nil, errors.NewTypedError(ErrCDAttribute, errors.New("require at least one attribute"))
 	}
 
-	ncd, err := cd.PrepareNewVersion(nil, CollaboratorsAccess{}, nil)
-	if err != nil {
-		return nil, errors.NewTypedError(ErrCDAttribute, errors.New("failed to prepare new version: %v", err))
+	var ncd *CoreDocument
+	var err error
+	if prepareNewVersion {
+		ncd, err = cd.PrepareNewVersion(nil, ca, nil)
```

Comment:
If you are updating an invoice, for example, and you prepare a new version and do not pass the prefix `invoice|po` that is used internally to prepare the transition rules, what is the consequence of that?

Reply:
the invoice|po should be passed else, the transition rules will not work for specific document